### DISCUSSION TITLE
Temp files location?

### DISCUSSION THREAD

**Discussion by thinkbig1979** [SpatiumPortae/portal#176] [Q&A]
Hi, I had to cancel a large transfer because it was taking ages to compress. I quit using the `q` key. 
I suspect think there's a temporary compressed file somewhere that needs cleaning up. Where would I find this file?
Thanks.

Oh and I agree with other comments regarding the ability to just start sending without compression. The compression step takes time and uses disk space and CPU, so being able to select a different method or disabling it altogether at will would be a welcome addition to this very useful tool :-)

**Comment by ZinoKader** [comment]
Hi!

Temp files should be cleaned up at the beginning of each new transfer, but if you feel like that did not work for you, the temp files are stored in the default temp file location for your OS, using `os.TempDir()`: https://pkg.go.dev/os#TempDir.

Here's the relevant text from the docs:

> On Unix systems, it returns $TMPDIR if non-empty, else /tmp. On Windows, it uses GetTempPath, returning the first non-empty value from %TMP%, %TEMP%, %USERPROFILE%, or the Windows directory. On Plan 9, it returns /tmp.

---

Regarding sending without compression, that would need some refactoring which breaks current functionality, i.e. for a v2. Unfortunately, both me and @mellonnen are kind of swamped right now. Contributions are very welcome, so if you feel like, just open a PR, I'm not very picky!

I appreciate and am really happy that you find the tool useful!
